src/jdk.incubator.vector/share/classes/jdk/incubator/vector/Float16.java

```diff
@@ -0,0 +1,1518 @@
+            return new Float16(sign_bit); // Positive or negative zero
+        }
+
+        // Dealing with finite values in exponent range of binary16
+        // (when rounding is done, could still round up)
+        int exp = Math.getExponent(d);
+        assert -25 <= exp && exp <= 15;
+
+        // For binary16 subnormals, beside forcing exp to -15, retain
+        // the difference expdelta = E_min - exp.  This is the excess
+        // shift value, in addition to 42, to be used in the
+        // computations below.  Further the (hidden) msb with value 1
+        // in d must be involved as well.
+        int expdelta = 0;
+        long msb = 0x0000_0000_0000_0000L;
+        if (exp < -14) {
+            expdelta = -14 - exp; // FIXME?
+            exp = -15;
+            msb = 0x0010_0000_0000_0000L; // should be 0x0020_... ?
+        }
+        long f_signif_bits = doppel & 0x000f_ffff_ffff_ffffL | msb;
+
+        // Significand bits as if using rounding to zero (truncation).
+        short signif_bits = (short)(f_signif_bits >> (42 + expdelta));
+
+        // For round to nearest even, determining whether or not to
+        // round up (in magnitude) is a function of the least
+        // significant bit (LSB), the next bit position (the round
+        // position), and the sticky bit (whether there are any
+        // nonzero bits in the exact result to the right of the round
+        // digit). An increment occurs in three cases:
+        //
+        // LSB  Round Sticky
+        // 0    1     1
+        // 1    1     0
+        // 1    1     1
+        // See "Computer Arithmetic Algorithms," Koren, Table 4.9
+
+        long lsb    = f_signif_bits & (1L << 42 + expdelta);
+        long round  = f_signif_bits & (1L << 41 + expdelta);
+        long sticky = f_signif_bits & ((1L << 41 + expdelta) - 1);
+
+        if (round != 0 && ((lsb | sticky) != 0 )) {
+            signif_bits++;
+        }
+
+        // No bits set in significand beyond the *first* exponent bit,
+        // not just the significand; quantity is added to the exponent
+        // to implement a carry out from rounding the significand.
+        assert (0xf800 & signif_bits) == 0x0;
+
+        return new Float16((short)(sign_bit | ( ((exp + 15) << 10) + signif_bits ) ));
+    }
+
+    /**
+     * Returns a {@code Float16} holding the floating-point value
+     * represented by the argument string.
+     *
+     * @implSpec
+     * The current implementation acts as if the string were
+     * {@linkplain Double#parseDouble(String) parsed} as a {@code
+     * double} and then {@linkplain #valueOf(double) converted} to
+     * {@code Float16}. This behavior is expected to change to
+     * accommodate the precision of {@code Float16}.
+     *
+     * @param  s the string to be parsed.
+     * @return the {@code Float16} value represented by the string
+     *         argument.
+     * @throws NullPointerException  if the string is null
```

Comment:
Should we move this NPE clause to the class specification, i.e. `Unless otherwise specified, all methods in this class throw {@code NullPointerException} when passed a {@code null}`? I think this applies to the APIs taking `BigDecimal` and static methods taking `Float16`. (Note we need to explicitly exclude `equals`)

Reply:
Hmm. I'm trying to keep parity between Float16 and java.lang.{Float, Double}. Float/Double don't include the class-level statement so I wouldn't include on in Float16 until they do. Writing Float16 has prompted several other improvements to Float/Double, so a null-handling statement is something worth considering.